pkg/cmd/traces.go

```diff
@@ -0,0 +1,115 @@
+package cmd
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"github.com/go-kit/log/level"
+	"github.com/observatorium/obsctl/pkg/config"
+	"github.com/spf13/cobra"
+)
+
+func NewTraceServicesCmd(ctx context.Context) *cobra.Command {
+	var outputFormat string
+	cmd := &cobra.Command{
+		Use:   "services",
+		Short: "List names of services",
+		Long:  "List names of services with trace information",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cfg, err := config.Read(logger)
+			if err != nil {
+				return fmt.Errorf("getting reading config: %w", err)
+			}
+
+			client, err := cfg.Client(ctx, logger)
+			if err != nil {
+				return fmt.Errorf("getting current client: %w", err)
+			}
+
+			level.Debug(logger).Log(
+				"msg", "Using configuration",
+				"URL", cfg.APIs[cfg.Current.API].URL,
+				"tenant", cfg.Current.Tenant)
+
+			url := fmt.Sprintf("%s%s", cfg.APIs[cfg.Current.API].URL, "api/traces/v1/rhobs/api/services")
+			resp, err := client.Get(url)
+			if err != nil {
+				return fmt.Errorf("getting: %w", err)
+			}
+			bodyBytes, err := ioutil.ReadAll(resp.Body)
+			if err != nil {
+				return fmt.Errorf("getting: %w", err)
+			}
+			if resp.StatusCode >= 300 {
+				fmt.Fprintf(os.Stdout, "%d: %s\n%s", resp.StatusCode, resp.Status, string(bodyBytes))
+				return fmt.Errorf("%d: %s", resp.StatusCode, resp.Status)
+			}
+
+			if outputFormat == "table" {
+				svcs, err := services(bodyBytes)
+				if err != nil {
+					return fmt.Errorf("parsing services: %w", err)
+				}
+				_ = printTable(cmd.OutOrStdout(), cmd.OutOrStderr(), svcs)
+			} else if outputFormat == "json" {
```

Comment:
I think we can simply default to `json` here, so users don't always need to pass the flag! 🙂 

Also, this block can be a switch-case one.

Reply:
We default to `table`.  The idea is to be like _kubectl_ or _cf_ CLIs, and default to table.  I will change it to a switch statement.